src-docs/src/views/comment/comment_example.js

```diff
@@ -0,0 +1,201 @@
+          code: commentSource,
+        },
+        {
+          type: GuideSectionTypes.HTML,
+          code: commentHtml,
+        },
+      ],
+      text: (
+        <div>
+          <p>
+            Use <strong>EuiComment</strong> for displaying comment threads with{' '}
+            <strong>EuiCommentList</strong>. Each <strong>EuiComment</strong>{' '}
+            has two parts: a <EuiCode>timelineIcon</EuiCode> on the left and
+            content on the right. The <EuiCode>timelineIcon</EuiCode> provides a
+            visual indication of the <EuiCode>type</EuiCode> of comment it is.
+            For example, it can be an icon that represents what action was
+            performed or it can be a user avatar. The content has a header with
+            all the relevant metadata and a body.
+          </p>
+        </div>
+      ),
+      props: { EuiComment },
+      snippet: commentSnippet,
+      demo: <Comment />,
+    },
+    {
+      title: 'Comment types',
+      source: [
+        {
+          type: GuideSectionTypes.JS,
+          code: commentTypesSource,
+        },
+        {
+          type: GuideSectionTypes.HTML,
+          code: commentTypesHtml,
+        },
+      ],
+      text: (
+        <div>
+          <p>
+            Use the default <EuiCode>type</EuiCode> of comment,{' '}
+            <EuiCode>regular</EuiCode> to display comments that a user has
+            written.
+          </p>
+          <p>
+            Use comments of type <EuiCode>update</EuiCode> to display comments
+            that generally do not have a body and are logging actions that
+            either the user or the system has performed (e.g. &ldquo;jsmith
+            edited a case&rdquo; or &ldquo;kibanamachine added the review
+            label&rdquo;).
+          </p>
+        </div>
+      ),
+      props: { EuiComment },
+      snippet: commentTypesSnippet,
+      demo: <CommentTypes />,
+    },
+    {
+      title: 'Custom timelineIcon',
+      source: [
+        {
+          type: GuideSectionTypes.JS,
+          code: commentTimelineIconsSource,
+        },
+        {
+          type: GuideSectionTypes.HTML,
+          code: commentTimelineIconsHtml,
+        },
+      ],
+      text: (
+        <div>
+          <p>
+            There are three ways to use <EuiCode>timelineIcon</EuiCode>:
+          </p>
+          <ol>
+            <li>
+              Use the defaults a) a user icon inside a 40x40 container for
+              comments of type
+              <EuiCode>regular</EuiCode> and b) a dot icon inside a 24x24
+              container for comments of type <EuiCode>update</EuiCode>.
+            </li>
+            <li>
+              Pass a string with any of the icon types that{' '}
+              <strong>EuiIcon</strong> supports and it will receive the default
+              styling.
+            </li>
+            <li>
+              Pass any other element (e.g. <strong>EuiAvatar</strong>). It is
```

Comment:
Can you also link to any external components mentioned? Like this would be:

```suggestion
              Pass any other element (e.g. <Link to="/display/avatar"><strong>EuiAvatar</strong></Link>). It is
```